addons/payment_stripe/static/src/js/express_checkout_form.js

```diff
@@ -0,0 +1,267 @@
+                        email: ev.payerEmail,
+                        phone: ev.payerPhone,
+                        street: ev.paymentMethod.billing_details.address.line1,
+                        street2: ev.paymentMethod.billing_details.address.line2 ?
+                            ev.paymentMethod.billing_details.address.line2 : false,
+                        zip: ev.paymentMethod.billing_details.address.postal_code,
+                        city: ev.paymentMethod.billing_details.address.city,
+                        country: ev.paymentMethod.billing_details.address.country,
+                        state: ev.paymentMethod.billing_details.address.state ?
+                            ev.paymentMethod.billing_details.address.state : false,
+                    }
+                }
+                if (isShippingInformationRequired) {
+                    addresses['shipping_address'] = {
+                        name: ev.shippingAddress.recipient,
+                        // email is a mandatory fields for a partner, so we use the same as the one
+                        // for the billing address
+                        email: ev.payerEmail,
+                        phone: ev.shippingAddress.phone,
+                        street: ev.shippingAddress.addressLine[0],
+                        street2: ev.shippingAddress.addressLine[1] ?
+                            ev.shippingAddress.addressLine[1]: false,
+                        zip: ev.shippingAddress.postalCode,
+                        city: ev.shippingAddress.city,
+                        country: ev.shippingAddress.country,
+                        state: ev.shippingAddress.region ? ev.shippingAddress.region : false,
+                    }
+                    addresses['shipping_option'] = ev.shippingOption
+                }
+                // Send addresses
+                this.txContext.partnerId = parseInt(await this._rpc({
+                    route: '/shop/express', params: addresses,
+                }));
+                const {client_secret} = await this._rpc({
+                    // Create transaction
+                    route: this.txContext.transactionRoute,
+                    params: this._prepareTransactionRouteParams(
+                        acquirerData.provider, acquirerData.acquirerId
+                    ),
+                })
+                // Confirm the PaymentIntent without handling potential next actions (yet).
+                const {paymentIntent, error: confirmError} = await stripeJS.confirmCardPayment(
+                    client_secret,
+                    {payment_method: ev.paymentMethod.id},
+                    {handleActions: false}
+                );
+                if (confirmError) {
+                    // Report to the browser that the payment failed, prompting it to
+                    // re-show the payment interface, or show an error message and close
+                    // the payStripeExpress('fail');
```

Comment:
Do we not do anything on error?

Reply:
Currently, we don't indeed. It's not a big deal since the error is already dealt with by Apple Pay and Google Pay, the user just have to open again the form. I won't merge the comments like this if we decide not to handle those errors.